### PR TITLE
Make Account entropy non nullable.

### PR DESCRIPTION
**Pull request by christian-oudard**
Also standardize some names, internally and on the API.
by_account -> for_account

### CODE REVIEW

**Comment by eranrund** [comment]
Is this taking the stance that full-service only supports adding accounts when starting from entropy bytes?

**Review by sugargoat — APPROVED** [review]
Looks great!

**Comment by sugargoat** [comment]
@eranrund - Yeah, we discussed and decided it's a bug in the MobileCoin ecosystem overall if there is a way to export an account key without entropy bytes.

File: full-service/src/db/account.rs [hunk 1]
```diff
@@ -523,114 +497,10 @@ mod tests {
             account_id_hex
         };
         let account = Account::get(&account_id, &wallet_db.get_conn().unwrap()).unwrap();
-        let decoded_entropy = RootEntropy::try_from(account.entropy.unwrap().as_slice()).unwrap();
+        // let decoded_entropy = RootEntropy::try_from(&account.entropy).unwrap();
```
Comment (mfaulk) [inline full-service/src/db/account.rs:500]: Leftover?

**Review by mfaulk — APPROVED** [review]

**Pull request merged**